client/my-sites/upgrades/checkout-thank-you/google-voucher-details.jsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import debounce from 'lodash/debounce';
+
+/**
+ * Internal dependencies
+ */
+import localize from 'lib/mixins/i18n/localize';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import PurchaseDetails from 'components/purchase-detail';
+import PurchaseButton from 'components/purchase-detail/purchase-button';
+import TipInfo from 'components/purchase-detail/tip-info';
+import Dialog from 'components/dialog';
+import googleTermsAndConditions from './google-voucher-TC';
+import QuerySiteVouchers from 'components/data/query-site-vouchers';
+import {
+	assignSiteVoucher as assignVoucher
+} from 'state/sites/vouchers/actions';
+import { GOOGLE_CREDITS } from 'state/sites/vouchers/service-types';
+import {
+	getVouchersBySite,
+	getGoogleAdCredits
+} from 'state/sites/vouchers/selectors';
+
+class GoogleVoucherDetails extends Component {
+	constructor() {
+		super();
+
+		// bounds
+		this.onButtonClick = this.onButtonClick.bind( this );
+		this.onDialogCancel = this.onDialogCancel.bind( this );
+		this.onAgreeButton = this.onAgreeButton.bind( this );
+
+		// debounced
+		this.incrementStep = debounce( this.incrementStep, 300 );
+
+		this.state = {
+			showTermsAndConditions: false,
+			step: 0
```

Comment:
Why do we need both `step` and `showTermsAndConditions` in state? They are not independent of each other. `showTermsAndConditions` is only ever true when `step === 1`. I think we would be better off with a single `step` attribute in `state`, which uses text strings to describe the current state, like `INITIAL_STEP`, `TERMS_AND_CONDITIONS`, and `CODE_REDEEMED`